addons/account/models/account_payment_term.py

```diff
@@ -43,6 +43,21 @@ def _check_lines(self):
             if len(lines) > 1:
                 raise ValidationError(_('A Payment Term should have only one line of type Balance.'))
 
+    @api.onchange('line_ids')
+    def _onchange_line_ids(self):
+        if not self.line_ids or (self.line_ids and self.line_ids.mapped('value').count('balance') != 1):
+            raise UserError(_('You must have only one line with the Balance type.'))
+
+    @api.model
+    def compute_due_date(self, line, date_ref):
+        due_date = fields.Date.from_string(date_ref)
+        due_date += relativedelta(months=line.months)
+        due_date += relativedelta(days=line.days)
+        if line.end_month:
+            due_date += relativedelta(day=31)
```

Comment:
I'm surprised you don't have anything resetting the date to the first day of month in case of `end_month`. Am I missing something?

Reply:
No need with the combination of `relativedelta`'s `day` and `days` options. E.g.:
![image](https://user-images.githubusercontent.com/91618364/186450816-b9b43e3e-c66a-4b1f-b135-71769013ceef.png)